packages/pigeon/test/java_generator_test.dart

```diff
@@ -870,6 +870,32 @@ void main() {
             'Long output = api.add((xArg == null) ? null : xArg.longValue(), (yArg == null) ? null : yArg.longValue())'));
   });
 
+  test('host Object argType', () {
```

Comment:
These test names don't communicate what is being validated, so if they were to fail it's not clear to me we would know what that meant. A test name should in general describe what conceptual thing is being validated.  What is the important thing being checked here as relates to this PR?

Alternately, if this is meant to cover several aspects of a specific API generation, then the individual `expect` calls should be annotated (see https://github.com/flutter/packages/blob/master_archive/packages/pigeon/test/cpp_generator_test.dart#L114-L125 for example)